main/tests/server/src/com/google/refine/exporters/TsvExporterTests.java

```diff
@@ -193,6 +193,468 @@ public void exportTsvWithEmptyCells() {
+
+        try {
+            SUT.export(project, options, engine, writer);
+        } catch (IOException e) {
+            Assert.fail();
+        }
+
+        Assert.assertEquals(writer.toString(), "column0\tcolumn1\tcolumn2\n" +
+                "row0cell0\trow0cell1\trow0cell2\n" +
+                "row1cell0\t\"\"cell enclosed by (\") quote character\"\"\trow1cell2\n" +
+                "row2cell0\trow2cell1\trow2cell2\n");
+    }
+
+    @Test
+    public void exportTsvWithDQuote() {
+        CreateGrid(3, 3);
+
+        project.rows.get(1).cells.set(1, new Cell("cell containing (\") double quote character", null));
+
+        try {
+            SUT.export(project, options, engine, writer);
+        } catch (IOException e) {
+            Assert.fail();
+        }
+
+        Assert.assertEquals(writer.toString(), "column0\tcolumn1\tcolumn2\n" +
+                "row0cell0\trow0cell1\trow0cell2\n" +
+                "row1cell0\t\"cell containing (\") double quote character\"\trow1cell2\n" +
+                "row2cell0\trow2cell1\trow2cell2\n");
+    }
+
+    @Test
+    public void exportTsvWithStartQuote() {
+        CreateGrid(3, 3);
+
+        project.rows.get(1).cells.set(1, new Cell("\"cell starting with a double quote character", null));
+
+        try {
+            SUT.export(project, options, engine, writer);
+        } catch (IOException e) {
+            Assert.fail();
+        }
+
+        Assert.assertEquals(writer.toString(), "column0\tcolumn1\tcolumn2\n" +
+                "row0cell0\trow0cell1\trow0cell2\n" +
+                "row1cell0\t\"\"cell starting with a double quote character\"\trow1cell2\n" +
+                "row2cell0\trow2cell1\trow2cell2\n");
+    }
+
+    @Test
+    public void exportTsvWithEndQuote() {
+        CreateGrid(3, 3);
+
+        project.rows.get(1).cells.set(1, new Cell("cell ending with a double quote character\"", null));
+
+        try {
+            SUT.export(project, options, engine, writer);
+        } catch (IOException e) {
+            Assert.fail();
+        }
+
+        Assert.assertEquals(writer.toString(), "column0\tcolumn1\tcolumn2\n" +
+                "row0cell0\trow0cell1\trow0cell2\n" +
+                "row1cell0\t\"cell ending with a double quote character\"\"\trow1cell2\n" +
+                "row2cell0\trow2cell1\trow2cell2\n");
+    }
+
+    @Test
+    public void exportTsvWithQuoteNTab() {
+        CreateGrid(3, 3);
+
+        project.rows.get(1).cells.set(1, new Cell("cell containing (\") double quote and (\t) tab character", null));
+
+        try {
+            SUT.export(project, options, engine, writer);
+        } catch (IOException e) {
+            Assert.fail();
+        }
+
+        Assert.assertEquals(writer.toString(), "column0\tcolumn1\tcolumn2\n" +
+                "row0cell0\trow0cell1\trow0cell2\n" +
+                "row1cell0\t\"cell containing (\") double quote and (\t) tab character\"\trow1cell2\n"
```

Comment:
This test case seems to depart from the desired behaviour stated here:
https://github.com/OpenRefine/OpenRefine/issues/2071#issuecomment-515681043

Double quotes which appear in cells enclosed by double quotes should be escaped.

Reply:
> This test case seems to depart from the desired behaviour stated here: [#2071 (comment)](https://github.com/OpenRefine/OpenRefine/issues/2071#issuecomment-515681043)
> 
> Double quotes which appear in cells enclosed by double quotes should be escaped.

This test does not seems to depart from the original issue to me as it is mentioning that without escaping there is a double/duplicate occurrence of double quote character. However, in this case there is no duplicate occurrence of double quote character.

 Below is the output produced from openrefine in both scenarios for ready reference:-

Input: "cell enclosed by (") quote character"

Output produced without escaping: """cell enclosed by ("") quote character"""

Output produced with escaping: ""cell enclosed by (") quote character""

Regards,
Vani

Reply:
Hi @antoine2711 @wetneb 

Do I need to work any further on this issue.

Thanks & Warm Regards